src/test/resources/tck/openapi3/discriminator.json

```diff
@@ -74,5 +74,174 @@
         "valid": false
       }
     ]
+  },
+  {
+    "description": "discriminator with oneOf validation with ref",
+    "schema": {
+      "oneOf": [
+        {
+          "$ref": "#/definitions/A"
+        },
+        {
+          "$ref": "#/definitions/B"
+        }
+      ],
+      "discriminator": {
+        "propertyName": "id"
+      },
+      "definitions": {
+        "A": {
+          "type": "object",
+          "properties": {
+            "id": {
+              "type": "string"
+            },
+            "value_a": {
+              "type": "string"
+            }
+          },
+          "required": ["id", "value_a"]
+        },
+        "B": {
+          "type": "object",
+          "properties": {
+            "id": {
+              "type": "string"
+            },
+            "value_b": {
+              "type": "string"
+            }
+          },
+          "required": ["id"]
+        }                
+      }
+    },
+    "tests": [
+      {
+        "description": "should be valid with id A and property value_a",
+        "data": {
+          "id": "A",
+          "value_a": "a"
+        },
+        "valid": true
+      },
+      {
+        "description": "should be invalid with id A and property value_b",
+        "data": {
+          "id": "A",
+          "value_b": "b"
+        },
+        "valid": false
+      },
+      {
+        "description": "should be valid with id B and property value_b",
+        "data": {
+          "id": "B",
+          "value_b": "b"
+        },
+        "valid": true
+      },
+      {
+        "description": "should be invalid without id",
+        "data": {
+          "value_b": "b"
+        },
+        "valid": false
+      },
+      {
+        "description": "should be invalid with id A and without value_a",
```

Comment:
Can you add 2 test cases: one with the empty data (`"data": {}`) and one with only the id `B`

Reply:
Same below